app/code/Magento/Review/Ui/DataProvider/Product/ReviewDataProvider.php

```diff
@@ -79,24 +79,43 @@ public function getData()
     }
 
     /**
-     * {@inheritdoc}
-     * @since 100.1.0
+     * Returns prepared field name
+     *
+     * @param string $name
+     * @return string
      */
-    public function addFilter(\Magento\Framework\Api\Filter $filter)
+    private function getPreparedField(string $name): string
     {
-        $field = $filter->getField();
+        $preparedName = '';
 
-        if (in_array($field, ['review_id', 'created_at', 'status_id'])) {
-            $filter->setField('rt.' . $field);
+        if (in_array($name, ['review_id', 'created_at', 'status_id'])) {
+            $preparedName = 'rt.' . $name;
+        } elseif (in_array($name, ['title', 'nickname', 'detail'])) {
+            $preparedName = 'rdt.' . $name;
+        } elseif ($name === 'review_created_at') {
+            $preparedName = 'rt.created_at';
         }
 
-        if (in_array($field, ['title', 'nickname', 'detail'])) {
-            $filter->setField('rdt.' . $field);
-        }
+        return $preparedName ?: $name;
+    }
 
-        if ($field === 'review_created_at') {
-            $filter->setField('rt.created_at');
-        }
+    /**
+     * @inheritDoc
+     */
+    public function addOrder($field, $direction): void
```

Comment:
Since this is an `@api`, and it is extending a class that implements `DataProviderInterface`, for keeping backward compatibility, I believe we should leave the function signature as it is on the interface.